### PR TITLE
Update patch flag help message

### PR DESCRIPTION
**Pull request by redbaron**
It wasn't clear what patch flag is doing exactly, so I had to ask and got a very clear reply here https://github.com/vmware-tanzu/carvel-kapp/discussions/382#discussioncomment-1773079

### CODE REVIEW

File: pkg/kapp/cmd/app/deploy_flags.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func (s *DeployFlags) Set(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&s.IntoNamespace, "into-ns", "", "Place resources into namespace")
 	cmd.Flags().StringSliceVar(&s.MapNamespaces, "map-ns", nil, "Map resources from one namespace into another (could be specified multiple times)")
 
-	cmd.Flags().BoolVarP(&s.Patch, "patch", "p", false, "Add or update provided resources")
+	cmd.Flags().BoolVarP(&s.Patch, "patch", "p", false, "Add or update existing resources only, never delete any")
```
Comment (100mik) [inline pkg/kapp/cmd/app/deploy_flags.go:37]: I wonder if just "Add or update existing resources only" is sufficient, but this definitely makes the flags purpose a lot clearer.

**Review by 100mik — APPROVED** [review]
LGTM!

**Comment by redbaron** [comment]
I signed CLA a while ago, no idea why  clabot isn't picking it up.

**Comment by cppforlife** [comment]
hmm, vmwclabot is still not happy... trying to reach folks who operate it..

**Review by BennyThink — APPROVED** [review]

**Comment by renuy** [comment]
The CLA request is  under review. hoping to hear back from the assigned team soon.

**Pull request merged**